Source/WebKit/UIProcess/ios/forms/WKDateTimeInputControl.mm

```diff
@@ -205,12 +252,44 @@ - (NSString *)_sanitizeInputValueForFormatter:(NSString *)value
 
 - (void)_dateChanged
 {
+#if HAVE(UICALENDARSELECTIONWEEKOFYEAR)
+    if (_view.focusedElementInformation.elementType == WebKit::InputType::Week) {
+        RetainPtr dateFormatter = [self iso8601DateFormatterForCalendarView];
+        [_view updateFocusedElementValue:[dateFormatter stringFromDate:[[NSCalendar calendarWithIdentifier:NSCalendarIdentifierISO8601] dateFromComponents:[_selectionWeekOfYear selectedWeekOfYear]]]];
+        return;
+    }
+#endif
     RetainPtr dateFormatter = [self dateFormatterForPicker];
     [_view updateFocusedElementValue:[dateFormatter stringFromDate:[_datePicker date]]];
 }
 
+#if HAVE(UICALENDARSELECTIONWEEKOFYEAR)
+- (void)setWeekPickerToInitialValue
+{
+    NSCalendarUnit unitFlags = NSCalendarUnitYearForWeekOfYear | NSCalendarUnitWeekOfYear;
+
+    if (![_initialValue length]) {
+        [_selectionWeekOfYear setSelectedWeekOfYear:[[NSCalendar calendarWithIdentifier:NSCalendarIdentifierISO8601] components:unitFlags fromDate:[NSDate date]]];
+        [self _dateChanged];
+        return;
+    }
+
+    RetainPtr parsedDate = [[self iso8601DateFormatterForCalendarView] dateFromString:[self _sanitizeInputValueForFormatter:_initialValue.get()]];
+    RetainPtr<NSDateComponents> dateComponents = [[NSCalendar calendarWithIdentifier:NSCalendarIdentifierISO8601] components:unitFlags fromDate:parsedDate.get() ? parsedDate.get() : [NSDate date]];
+
+    // FIXME: The selectedWeekOfYear from these components causes the week before the correct one to be selected, so it is manually adjusted here. There is likely a better way of achieving this.
+    dateComponents.get().weekOfYear += 1;
```

Comment:
This seems a little concerning. Have you tested in device locales where the first weekday differs/matches from ISO8601?

In these cases we want to make sure what the user sees (both in-page, and the picker) matches the week number in their locale. This number could be different from what is exposed via JavaScript / HTML value.